src/ray/raylet/node_manager.cc

```diff
@@ -2897,25 +2897,52 @@ MemoryUsageRefreshCallback NodeManager::CreateMemoryUsageRefreshCallback() {
             << "worker pid: " << high_memory_eviction_target_->GetProcess().GetId()
             << "task: " << high_memory_eviction_target_->GetAssignedTaskId();
       } else {
-        auto newest_task_start_time = std::chrono::high_resolution_clock::time_point();
-        std::shared_ptr<WorkerInterface> latest_worker = nullptr;
-        for (const auto &worker : worker_pool_.GetAllRegisteredWorkers()) {
-          if (worker->GetAssignedTaskTime() > newest_task_start_time) {
-            newest_task_start_time = worker->GetAssignedTaskTime();
-            latest_worker = worker;
+        auto workers = worker_pool_.GetAllRegisteredWorkers();
+        if (!workers.empty()) {
+          std::sort(workers.begin(),
+                    workers.end(),
+                    [](std::shared_ptr<WorkerInterface> const &left,
+                       std::shared_ptr<WorkerInterface> const &right) -> bool {
+                      return left->GetAssignedTaskTime() > right->GetAssignedTaskTime();
+                    });
+          std::shared_ptr<WorkerInterface> latest_worker = workers.front();
+          std::ostringstream top_n_latest_workers_info;
+          int64_t top_n = 0;
+
+          const static int64_t max_to_print = 10;
+          for (auto &worker : workers) {
+            auto pid = worker->GetProcess().GetId();
+            auto used_memory = this->memory_monitor_->GetProcessMemoryBytes(pid);
+            top_n_latest_workers_info
+                << "Worker " << top_n << " task assigned time counter "
+                << worker->GetAssignedTaskTime().time_since_epoch().count()
+                << " memory used " << used_memory << " task spec "
+                << worker->GetAssignedTask().GetTaskSpecification().DebugString() << "\n";
```

Comment:
Can we make sure that this information is concise andunderstandable to the user? I'm afraid this content is more targeted to a Ray core developer than a user.

Basically we want to include the minimum information that they need to go get the right logs and figure out what happened. You can take a look at the Python-based memory monitor to see what kind of output we should include.

Reply:
Off the top of my head, I'm thinking:
- memory used (human-readable units)
- task name (to connect to the application)
- maybe how long the task ran for before it got killed? Since the time counter is not meaningful to a user.
- task and worker ID + the CLI command they can use to go get the logs